pkg/telemetry/sqlite_metrics_db.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package telemetry
+
+import (
+	"database/sql"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+
+	// Import the sqlite3 driver
+	_ "modernc.org/sqlite"
+
+	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/common"
+)
+
+// sqliteMetricsDB implements the MetricDB interface using SQLite database
+type sqliteMetricsDB struct {
+	// metricsDBFile represents the full path to the SQLite DB file
+	metricsDBFile string
+}
+
+const (
+	// SQliteDBFileName is the name of the DB file that has CLI metrics
+	SQliteDBFileName = "cli_metrics.db"
+
+	// TanzuCLITelemetryMaxRowCount Max metric instances to be accumulated before pausing the collection
+	TanzuCLITelemetryMaxRowCount = 1000
+
+	// cliOperationMetricRowClause is the SELECT section of the SQL query to be used when querying the Metric DB row count.
+	cliOperationMetricRowClause = "SELECT count(*) FROM tanzu_cli_operations"
+)
+
+// Structure of each row of the PluginBinaries table within the SQLite database
+type cliOperationsRow struct {
+	cliVersion         string
+	osName             string
+	osArch             string
+	pluginName         string
+	pluginVersion      string
+	command            string
+	cliID              string
+	commandStartTSMsec string
+	commandEndTSMsec   string
+	cspOrgID           string
+	accountNumber      string
+	target             string
+	nameArg            string
+	endpoint           string
+	flags              string
+	exitStatus         int
+	isInternal         bool
+	error              string
+}
+
+// NewSQLiteMetricsDB returns a new PluginInventory connected to the data found at 'metricsDBFile'.
+func NewSQLiteMetricsDB() MetricsDB {
```

Comment:
Is there any reason we are making it public? as we are hard-coding the Telemetry SQlite DB file name, i would not make it public! If we are passing the path as input then maybe we can make it public!

Reply:
I will update it to private as we accessing it from the same package, we can make it public if this is moved to different package under telemetry in future.